_posts/2021/2021-03-31-hmat_memkind.md

```diff
@@ -0,0 +1,189 @@
+
+### What's new 
+
+**Locality**
+
+So far in memkind, we determined the locality as the nearest NUMA based on the shortest NUMA distance metrics. 
+In this strategy, we miss information about Socket and NUMA Node relation. In the case of non-symmetrical memory topology,
+where, for example, PMEM is placed only on one socket, and we would like to avoid remote memory access we could use
+one of the memory kinds from the new API to allocate from memory in local NUMA domain.
+
+<figure class="image">
+  <img src="/assets/memkind_single_pmem_medium_view.png" alt="non-symetric-medium-view">
+  <figcaption>Figure 5. Memory architecture overview based on physical medium.</figcaption>
+</figure>
+
+Using `MEMKIND_HIGHEST_CAPACITY_LOCAL` memkind will allocate to highest capacity Node inside a socket:
+
+<figure class="image">
+  <img src="/assets/memkind_single_pmem_property_view_alloc.png" alt="non-symetric-property-view">
+  <figcaption>Figure 6. Memory architecture overview based on memory property.</figcaption>
+</figure>
+
+**New API**
+
+Memory property like bandwidth and latency depends on the specific initiator of the memory request. This results in
+multidimensional analysis based on the different NUMA Node relationships and [performance characteristics][numa-performance].
+Based on this fact, with the new API, memkind addresses different scenarios of potential usage in more complicated memory topologies.
+
+**High Bandwidth**
+
+To help using legacy API, for `MEMKIND_HBW*` kinds, we introduced `MEMKIND_HBW_THRESHOLD` environment variable.
+Platforms may have multiple different types of memory, and each of them offers various bandwidth to the initiator of the memory request.
+Any NUMA Node with `read bandwidth` memory property (reported by node’s performance characteristics) above the **specific** threshold
+is considered as high bandwidth. Memkind identify these NUMA Node(s) as a `High Bandwidth` Memory NUMA Node(s).
+ `MEMKIND_HBW_THRESHOLD` provides a mechanism to define the specific value of this threshold.
+
+### What’s Coming Next
+
+The idea of NUMA nodes has evolved, and it's much different now compared to the past version. It no longer focuses on
+local and remote access. The modern approach is based on a concept with the initiator of a memory request and
+target of a memory request. Both initiator and target are characterized by different performance properties. 
+CPU is no longer the only entity which can make a memory access. And the same follows for memory  – if something performed a memory access,
+it doesn't have to come from DRAM.
+
+We believe that more complex memory architectures will become increasingly popular in the future.
+
+With this perspective, a one-dimensional approach based on NUMA distance metrics needs to be revised.
+Hardware/Firmware architects already defined solutions with HMAT, which means that HMAT-aware platforms are expected to appear in the future.
```

Comment:
Maybe it is worth to add something like: "at the time of writing" to not confuse readers in the future when such platforms will be available.

Reply:
Done